gotham/src/http/response/mod.rs

```diff
@@ -69,6 +70,84 @@ pub fn create_response(state: &State, status: StatusCode, body: Option<Body>) ->
     res
 }
 
+/// Produces a simple empty `Response` with a `Location` header.
+///
+/// This function is intended as a convenience. For more elaborate redirection
+/// needs, consider constructing a response from scratch.
+///
+/// # Examples
+///
+/// For the basic case where you simply want to forward a request to a
+/// different URI, `create_simple_rectirect()` may be all you need.
+///
+/// ```rust
+/// # extern crate gotham;
+/// # extern crate hyper;
+/// #
+/// # use hyper::{Response, StatusCode};
+/// # use gotham::state::State;
+/// # use gotham::http::response::create_simple_redirect;
+/// # use gotham::test::TestServer;
+/// # use hyper::header::Location;
+/// fn handler(state: State) -> (State, Response) {
+///     // permanent=true for a 301 (permanent) redirect
+///     // set to false for a 302 (temporary)
+///     let resp = create_simple_redirect(&state, true, "/over-there");
+///
+///     (state, resp)
+/// }
+/// # fn handler2(state: State) -> (State, Response) {
+/// #     let resp = create_simple_redirect(&state, false, "/detour");
+/// #
+/// #     (state, resp)
+/// # }
+/// # fn main() {
+/// #     {
+/// #         let test_server = TestServer::new(|| Ok(handler)).unwrap();
+/// #         let response = test_server
+/// #             .client()
+/// #             .get("http://example.com/")
+/// #             .perform()
+/// #             .unwrap();
+/// #
+/// #         assert_eq!(response.status(), StatusCode::PermanentRedirect);
+/// #         assert_eq!(
+/// #             response.headers().get::<Location>(),
+/// #             Some(&Location::new("/over-there"))
+/// #         );
+/// #     }
+/// #     {
+/// #         let test_server = TestServer::new(|| Ok(handler2)).unwrap();
+/// #         let response = test_server
+/// #             .client()
+/// #             .get("http://example.com/")
+/// #             .perform()
+/// #             .unwrap();
+/// #
+/// #         assert_eq!(response.status(), StatusCode::TemporaryRedirect);
+/// #         assert_eq!(
+/// #             response.headers().get::<Location>(),
+/// #             Some(&Location::new("/detour"))
+/// #         );
+/// #     }
+/// # }
+/// ```
+pub fn create_simple_redirect<L: Into<Cow<'static, str>>>(
```

Comment:
I'm thinking we should have explicit functions here as opposed to giving a bool so much 'power' that would be very easy to trip over.

`create_temporary_redirect` and `create_permanent_redirect` would be my suggestion for the public API (utilising the same private function internally for that DRY goodness).